flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -521,14 +521,14 @@ private void runMailboxLoop() throws Exception {
 	private void afterInvoke() throws Exception {
 		LOG.debug("Finished task {}", getName());
 
+		// close all operators in a chain effect way. closing happens from head to tail
+		// operator in the chain, contrary to StreamOperator#open() which happens tail to head.
+		// (see #initializeStateAndOpen()).
+		operatorChain.getHeadOperatorWrapper().close(this, true);
```

Comment:
We're passing `StreamTask.this` here to only synchronize on its `checkpointLock`, which was deprecated.
Can we pass and use `actionExecutor` instead?